drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+
+
+def check_rules_directory_exist(path: Path) -> bool:
+    # this method checks whether a non-empty directory exists
+    return path.is_dir() and any(path.iterdir())
+
+
+def find_process_in_pstree(pstree: List, pid: int, procname: str) -> Dict:
+    # this methods searches for a process in the generated process_tree.json file
+    for process in pstree:
+        if process["pid"] == pid:
+            return process
+        else:
+            child_processes = find_process_in_pstree(process["children"], pid, procname)
+            if child_processes:
+                return child_processes
+
+    # return None in case the process is not found
+    return None
+
+
+def get_all_child_processes(process: Dict) -> Iterator[int]:
+    # this method returns all of the child processes pids (recursively), including their parent's
+    for p in process["children"]:
+        yield from get_all_child_processes(p)
+
+    # yield current node's pid
+    yield process["pid"]
+
+
+def get_rules(rules_dir: List[Path]) -> Optional[capa.rules.RuleSet]:
+    try:
+        rules = capa.rules.get_rules(rules_dir)
+    except (IOError, capa.rules.InvalidRule, capa.rules.InvalidRuleSet):
+        # display the official capa error message in case there exists any malformed capa rules
+        rules = None
+        logger.exception(
+            "Make sure your file directory contains properly formatted capa rules. You can download the standard "
+            + "collection of capa rules from https://github.com/mandiant/capa-rules/releases."
+        )
+        logger.error(
+            "Please ensure you're using the rules that correspond to your major version of capa (%s)",
+            capa.version.get_major_version(),
+        )
+        logger.error(
+            "Or, for more details, see the rule set documentation here: %s",
+            "https://github.com/mandiant/capa/blob/master/doc/rules.md",
+        )
+
+    return rules
+
+
+def filter_rules(
+    rules: capa.rules.RuleSet, filter_function=lambda rule: rule
+) -> capa.rules.RuleSet:
+    # this method filters the given rules folder following arbitrary logic (specified by filter_function)
+    rules = list(rules.rules.values())
+    reduced_rules = list(filter(filter_function, rules))
+
+    # filter the given rules folder using the specified boolean function
+    filtered_rules = set()
+    for rule in reduced_rules:
+        filtered_rules.update(
+            set(capa.rules.get_rules_and_dependencies(rules, rule.name))
+        )
+
+    # generate the new ruleset, and return it
+    return capa.rules.RuleSet(list(filtered_rules))
+
+
+def get_malware_processes(
+    metadata_path: Path, inject_path: Path, pstree_path: Path
+) -> Optional[List[str]]:
+    # this method gets all the pids in the Drakvuf report that are associated with malware
+    with metadata_path.open("r") as f:
+        # we use the metadata file to get the analysis' start command
+        metadata = orjson.loads(f.read())
```

Comment:
(as per comment below, metadata parsing can be probably removed from this method)